framework/sandstone.cpp

```diff
@@ -805,13 +805,22 @@ static void init_internal(const struct test *test)
 
 static void initialize_smi_counts()
 {
-    std::optional<uint64_t> v = sApp->count_smi_events(cpu_info[0].cpu_number);
+    // iterate backwards because usually the highest cpu_numbers are found last
+    int i = num_cpus() - 1;
+    int cpu = cpu_info[i].cpu_number;
+    std::optional<uint64_t> v = sApp->count_smi_events(cpu);
     if (!v)
         return;
-    sApp->smi_counts_start.resize(num_cpus());
-    sApp->smi_counts_start[0] = *v;
-    for (int i = 1; i < num_cpus(); i++)
-        sApp->smi_counts_start[cpu_info[i].cpu_number] = sApp->count_smi_events(cpu_info[i].cpu_number).value_or(0);
+
+    for (;;) {
+        if (cpu >= sApp->smi_counts_start.size())
+            sApp->smi_counts_start.resize(cpu + 1);
+        sApp->smi_counts_start[cpu] = v.value_or(0);
```

Comment:
MSI_COUNT MSR has Core/Thread scope. Shouldn't we read it for each `cpu`, as we did before?

Reply:
That didn't change; compare to the current main branch version: https://github.com/opendcdiag/opendcdiag/blob/8dc3fc651552be86a34f749d3f7aed2c5567656d/framework/sandstone.cpp#L735-L739

But you're right, this doesn't need to store on the system processor number. Will fix.